app/controllers/chargeback_controller.rb

```diff
@@ -390,47 +390,39 @@ def cb_rpts_fetch_saved_report(id)
       return
     end
     @right_cell_text ||= _("Saved Chargeback Report [%{name}]") % {:name => rr.name}
-    if rr.userid != session[:userid]
+    if !current_user.miq_group_ids.include?(rr.miq_group_id) && !admin_user?
       add_flash(_("Report is not authorized for the logged in user"), :error)
       @saved_reports = cb_rpts_get_all_reps(id.split('-')[1])
       return
     else
       @report_result_id = session[:report_result_id] = rr.id
       session[:report_result_runtime]  = rr.last_run_on
-      @report = rr.report_results
-      session[:rpt_task_id] = nil
-      if @report.blank?
-        add_flash(_("Saved Report \"%{name}\" not found, Schedule may have failed") %
-          {:name => format_timezone(rr.last_run_on, Time.zone, "gtl")}, :error)
-        @saved_reports = cb_rpts_get_all_reps(rr.miq_report_id.to_s)
-        rep = MiqReport.find_by_id(rr.miq_report_id)
-        if x_active_tree == :cb_reports
-          self.x_node = "reports-#{rep.id}"
+      if rr.status.casecmp("finished") == 0
+        @report = rr.report_results
+        session[:rpt_task_id] = nil
+        if @report.blank?
+          add_flash(_("Saved Report \"%{name}\" not found, Schedule may have failed") %
+            {:name => format_timezone(rr.last_run_on, Time.zone, "gtl")}, :error)
+          @saved_reports = cb_rpts_get_all_reps(rr.miq_report_id.to_s)
+          rep = MiqReport.find_by_id(rr.miq_report_id)
+          if x_active_tree == :cb_reports_tree
+            self.x_node = "reports-#{rep.id}"
+          end
+          return
         else
-          @sb[:rpt_menu].each_with_index do |lvl1, i|
```

Comment:
@gtanzillo I believe that this `each_with_index` is never used because I think that it is copied from general saved reports
 https://github.com/ManageIQ/manageiq/blob/master/app/controllers/report_controller/saved_reports.rb#L40
and in the saved reports there are section like **saved report(1)** and list of **reports with saved reports and custom reports (2)**  (see picture)
and in chargeback controller It seems that `@sb[:rpt_menu]` isn't never filled instead of general saved reports
https://github.com/ManageIQ/manageiq/blob/master/app/controllers/report_controller.rb#L636
so in general saved report `@sb[:rpt_menu]`  is for (2) 

**So reasons to remove it from chargeback controller:**
- it causes error when I wanted to access saved report but without any records 
- and we don't have menu (2) in chargeback controller so I think that it can removed.

![screen shot 2016-05-25 at 16 13 13 1](https://cloud.githubusercontent.com/assets/14937244/15543188/38a4fb2e-2294-11e6-91e3-9d60886941f0.png)

So due this facts I think that is it ok to remove but can  @h-kataria or somebody from UI confirm it, please ? it would be better 👍 

thanks!